core-bundle/src/Resources/contao/models/PageModel.php

```diff
@@ -327,6 +327,10 @@ public function __set($strKey, $varValue)
 					case 'robots':
 						$htmlHeadBag->setMetaRobots($varValue);
 						break;
+
+					case 'noSearch':
+						$htmlHeadBag->setMetaRobots($varValue ? 'noindex,nofollow' : 'index,nofollow');
```

Comment:
Shouldn’t this modify the JSON-LD `noSearch` for schema.contao.org/Page instead of setting the meta robots?

Reply:
It would be way more complicated so I decided against it. Especially because the use case to set `noSearch = true` has never been "disable the internal search only". It has always been "disable search indexing". All the cases I've encountered where we used `->noSearch` (which are all gone in core now so the use case is questionable anyway), we wanted to exempt the response from being indexed by **any** search engine. 
Hence, my decision to not go down the complex road. It's not worth it.

Reply:
I have to disagree here, we specifically distinct in the page settings between robots and Contao's internal search engine, i.e. you can set a page to `noindex`, but still allow it to be indexed by Contao's search index and you can set it to `index` and disallow the page to be indexed by Contao's search index.

Reply:
I’m pretty sure that setting all pages that match one of the `TL_NOINDEX_KEYS` to `noindex,nofollow` would have very bad effects for SEO.

Reply:
It's changed now.